kube-runtime/src/reflector/object_ref.rs

```diff
@@ -44,6 +48,24 @@ pub struct ObjectRef<K: Resource> {
     /// assert_ne!(ObjectRef::<ConfigMap>::new("foo"), ObjectRef::new("foo").within("bar"));
     /// ```
     pub namespace: Option<String>,
+    /// Extra information about the object being referred to
+    ///
+    /// This is *not* considered when comparing objects, but may be used when converting to and from other representations,
+    /// such as [`OwnerReference`] or [`ObjectReference`].
+    #[derivative(Hash = "ignore", PartialEq = "ignore")]
+    pub extra: Extra,
+}
+
+/// Non-vital information about an object being referred to
+///
+/// See [`ObjectRef::extra`].
+#[derive(Default, Debug, Clone)]
+#[non_exhaustive]
```

Comment:
does `non_exhaustive` here technically not technically make this struct a bit less useful than `ObjectReference`?

Reply:
`#[non_exhaustive]` forces users to use our constructors (`ObjectRef::new`, `ObjectRef::from_obj`) rather than constructing new objects from scratch. That leaves us more open to adding new fields down the line.

Reply:
ah, the constructors still work of course :facepalm:
yup, sensible then!